modules/external-api/src/main/java/org/opencastproject/external/endpoint/SeriesEndpoint.java

```diff
@@ -230,18 +230,17 @@ public Response getSeriesList(@HeaderParam("Accept") String acceptHeader, @Query
           } else if ("contributors".equals(name)) {
             query.withContributor(value);
           } else if ("CreationDate".equals(name)) {
-            if (name.split("/").length == 2) {
               try {
-                Tuple<Date, Date> fromAndToCreationRange = getFromAndToCreationRange(name.split("/")[0],
-                        name.split("/")[1]);
+                Tuple<Date, Date> fromAndToCreationRange = getFromAndToCreationRange(value.split("/")[0],
+                        value.split("/")[1]);
                 query.withCreatedFrom(fromAndToCreationRange.getA());
                 query.withCreatedTo(fromAndToCreationRange.getB());
               } catch (IllegalArgumentException e) {
                 return RestUtil.R.badRequest(e.getMessage());
+              } catch (ArrayIndexOutOfBoundsException e) {
+                logger.error("Malformed date period. Correct UTC time period format: yyyy-MM-ddTHH:mm:ssZ/yyyy-MM-ddTHH:mm:ssZ");
+                return RestUtil.R.badRequest(e.getMessage());
```

Comment:
It would be useful, but not critical, to have that Malformed date period message in the return message instead of the Index out of bounds message.
For the log, that's great but maybe at INFO or WARN instead of ERROR. I view ERROR as a code bug problem versus a usage problem. Just an opinion. Feel free to argue against this comment.

Reply:
Thanks for your reply, Karen, I just made the adjustments. It's true, when the API call has a user error should be warning, instead Error in the server log. Also, I added the error message to the API.